core/stdlib/std.ncl

```diff
@@ -1378,9 +1379,15 @@
               else
                 value
             in
-            DependentFun
-              ArrayIndexFirst
-              (fun index => ArrayIndexSecond index -> Dyn),
+            let contract
+              | [| 'Index, 'Split |] -> Dyn -> Dyn
```

Comment:
In general we've refrained from contract annotations within the definition of a contract, although this one is rather harmless. That being said, if we want to add a contract, any reason to attach it here, instead of hoisting it up to `IndexedArrayFun`, so that it's visible in the documentation and the blame error tells that `IndexedArrayFun` is the culprit?

Reply:
My bad, that's not how I usually annotate contract either... moved to the top.